lib/topological_inventory/schema/default.rb

```diff
@@ -20,6 +20,7 @@ def initialize_inventory_collections
         add_default_collection(:subscriptions)
         add_default_collection(:vms)
         add_cross_link_vms
+        add_tags
```

Comment:
so we should be able to create records in taggings table directly so modeling tagging IC? Adding also the tags IC.

In OpenShift we did taggings per each parent IC like  https://github.com/Ladas/manageiq-providers-kubernetes/blob/9eb61a16c5cdbbe22b72fb3de74e173cdab32935/app/models/manageiq/providers/kubernetes/inventory/persister/definitions/container_collections.rb#L176

Then we can just do:
```ruby
collections['taggings_for_vms'].build(:tag => collections['tags].lazy_find('super_tag'), :taggable => collections['vms'].lazy_find(XY))
```

And you don't need to worry about ```:tag => collections['tags].lazy_find('super_tag')```, since we just have :name in that table, so we'll just do skeletal precreate. (not sure why is there ```tagging_count``` in that table, that looks unrelated to multi tenant world, we should just delete that column)

Reply:
we'll need to enhance the unique indexes though https://github.com/ManageIQ/topological_inventory-core/blob/master/db/schema.rb#L411

look e.g. at hardwares here https://github.com/ManageIQ/manageiq-schema/pull/19/files#diff-5214f0a7b9057034ce42bbc155d56284R86

in the current setup, you'll need also an index that where the tagger cols and context are nil? We'll not be filling those, correct? If that is the case, maybe we should delete these cols?

Reply:
we want to create ```tags``` dynamically, right? Btw. the other alternative would be to store in into jsonb, I could check if we can index those. Wouldn't that be better?

Reply:
Hm why would we need a different IC per parent if we can just set the taggable type/id?  That would make it easier to tag anything without having double the ICs (one for the actual collection and one for the taggigns)?

Reply:
> (not sure why is there tagging_count in that table, that looks unrelated to multi tenant world, we should just delete that column)

Yeah agreed, I want to leave the schema the same though until we decide if we're dropping acts-as-taggable-on or just not using it to set tags from inventory refresh.

Reply:
> we want to create tags dynamically, right?

Right

> Btw. the other alternative would be to store in into jsonb, I could check if we can index those. Wouldn't that be better?

Possibly?  It depends on how we're using the tags, would be nice to get this working then when we know how we're going to use the tags on the calling side do some perf tests.

Reply:
> so we should be able to create records in taggings table directly so modeling tagging IC? Adding also the tags IC.

Oh yeah, and I did add both taggings and tags ICs https://github.com/ManageIQ/topological_inventory-core/pull/67/files#diff-bf581c67a3bfa3b364180f998a222a83R23

Reply:
> Hm why would we need a different IC per parent if we can just set the taggable type/id? That would make it easier to tag anything without having double the ICs (one for the actual collection and one for the taggigns)?

The query is build from source. So e.g. source joins VM that joins tagging (where the relation has the right type baked in, so we can do the join)

If you'd want to do 1 query for e.g. VM and Network, we'd need to do union of 2 queries. And that query would get expensive, for all taggable objects.

Using just taggable type/id for filtering. Right now, we have no way do define the query, to ensure it stays under 1 scope (well we could rely on the fact the lazy_finds are bound to the scope). And no way to e.g. build a query to give you the all tags of list of entities, so you can delete the inactive ones. Well we'd have to build custom saver to attempt this.